plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,312 @@
+ * TODO sometimes convertView is a final param in getView.
+ * Possible solution: remove final modifier
+ */
+
+/** See {@link #getDescription()} method. */
+public class AndroidViewHolderRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Optimization for Android applications to optimize getView routines. "
+                + "It allows reducing the calls to inflate and getViewById Android " + "API methods.";
+    }
+
+    @Override
+    public String getName() {
+        return "Android ViewHolder";
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        IMethodBinding methodBinding = node.resolveBinding();
+        Block body = node.getBody();
+        if (body != null && isMethod(methodBinding, "android.widget.Adapter", "getView", "int", "android.view.View",
+                "android.view.ViewGroup")) {
+            GetViewVisitor visitor = new GetViewVisitor();
+            body.accept(visitor);
+            if (visitor.viewVariable != null && !visitor.isInflateInsideIf()) {
+                final ASTBuilder b = this.ctx.getASTBuilder();
+                final Refactorings r = this.ctx.getRefactorings();
+
+                // Transform tree
+
+                // Create If statement
+                // test-clause
+                InfixExpression condition = b.infixExpr(b.simpleName("convertView"), EQUALS, b.null0());
+                // then
+                Assignment assignment = b.assign(b.simpleName("convertView"), ASSIGN, b.copy(visitor.getInflateExpr()));
+                Block thenBlock = b.block(b.toStmt(assignment));
+
+                IfStatement ifStmt = b.if0(condition, thenBlock);
+                r.insertBefore(ifStmt, visitor.viewAssignmentStmt);
+
+                // assign to local view variable when necessary
+                if (!"convertView".equals(visitor.viewVariable.getIdentifier())) {
+                    Statement assignConvertViewToView = null;
+                    if (visitor.viewVariableDeclarationFragment != null) {
+                        final TypeNameDecider typeNameDecider = new TypeNameDecider(visitor.viewVariable);
+                        assignConvertViewToView = b.declareStmt(b.copyType(visitor.viewVariable, typeNameDecider),
+                                b.copy(visitor.viewVariable), b.simpleName("convertView"));
+                    } else if (visitor.viewVariableAssignment != null) {
+                        assignConvertViewToView = b
+                                .toStmt(b.assign(b.copy(visitor.viewVariable), ASSIGN, b.simpleName("convertView")));
+                    }
+                    if (assignConvertViewToView != null) {
+                        r.insertBefore(assignConvertViewToView, visitor.viewAssignmentStmt);
+                    }
+                }
+
+                // make sure method returns the view to be reused DELETEME
```

Comment:
Indeed this code does not change the result for the tests.
Can it be safely removed?